### PR TITLE
User/naethell/api versions

### PR DESCRIPTION
**Pull request by natalieethell**
added apiVersions and support for using a secondary test account (specifically for mountpoint tests)

### CODE REVIEW

**Comment by natalieethell** [comment]
@daspek let me know what you think of the changes!

**Comment by daspek** [comment]
i don't see anything here, but i noticed in the demo awhile back that you overloaded scopes to also represent service capabilities. i'd rather just add another variable for that called 'tags' or something that works largely the same way, but is separate from the concept of auth scopes

File: ApiDocs.Validation/MethodDefinition.cs [hunk 1]
```diff
@@ -165,13 +172,13 @@ public void AddTestParams(string rawContent)
         /// <param name="documents"></param>
         /// <param name="account"></param>
         /// <returns></returns>
-        public async Task<ValidationResult<HttpRequest>> GenerateMethodRequestAsync(ScenarioDefinition scenario, DocSet documents, IServiceAccount account)
+        public async Task<ValidationResult<HttpRequest>> GenerateMethodRequestAsync(ScenarioDefinition scenario, DocSet documents, IServiceAccount[] account)
```
Comment (chackman) [inline ApiDocs.Validation/MethodDefinition.cs]: Will account ever have more than 2 accounts (primary and secondary)?
If at most there are the primary and secondary accounts, I would prefer that those accounts be named explicitly.  It makes the code cleaner.

Comment (chackman) [reply ApiDocs.Validation/MethodDefinition.cs]: example: `IServiceAccount primaryAccount, IServiceAccount secondaryAccount`

Comment (natalieethell) [reply ApiDocs.Validation/MethodDefinition.cs]: For now, I don't think we'll ever need more than two accounts. Sounds good, I'll change that

File: ApiDocs.Validation/MethodDefinition.cs [hunk 2]
```diff
@@ -185,13 +199,28 @@ public void AddTestParams(string rawContent)
                     {
                         try
                         {
-                            var result = await setupRequest.MakeSetupRequestAsync(storedValuesForScenario, documents, scenario, account);
-                            errors.AddRange(result.Messages);
-
-                            if (result.IsWarningOrError)
+                            // use secondary account for specific setup requests
+                            if (setupRequest.SecondaryAccount && secondaryAccount != null)
```
Comment (chackman) [inline ApiDocs.Validation/MethodDefinition.cs]: Please warn if `SecondaryAccount` is true but `secondaryAccount` is null.

**Comment by natalieethell** [comment]
Currently looking into a change that might need to be made to support secondary accounts in ApiDocs.Console

File: ApiDocs.Console/Program.cs [hunk 3]
```diff
@@ -966,6 +966,14 @@ internal static void WriteValidationError(string indent, ValidationError error)
                 return false;
             }
 
+            IServiceAccount secondaryAccount = null;
+            if (options.SecondaryAccountName != null)
+            {
+                var secondaryAccounts = options.FoundAccounts.Where(
+                    x => options.SecondaryAccountName.Equals(x.Name));
+                secondaryAccount = secondaryAccounts.Any() ? secondaryAccounts.First() : null;
```
Comment (chackman) [inline ApiDocs.Console/Program.cs]: Assuming `null` is the default value, this would be more concise: `secondaryAccounts.FirstOrDefault()`

**Pull request merged**